### PR TITLE
[BEAM-13911] Add basic tests to Go direct runner.

### PR DESCRIPTION
**Pull request by lostluck**
Copy tests from my direct runner replacement project to add coverage to the direct runner. This handles a number of features tested elsewhere in the SDK, but it overall hurts our coverage number as a result. Adding a suite of simple pipelines against the Direct runner improves it's coverage by ~70%, even without touching the Combine and SDF edge cases.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @jrmccluskey @riteshghorse @damccorm

File: sdks/go/pkg/beam/runners/direct/direct_test.go [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+	sort.Ints(fn.Want)
+	if d := cmp.Diff(fn.Want, fn.got); d != "" {
+		return fmt.Errorf("int64Check[%v] (-want, +got): %v", fn.Name, d)
+	}
+	return nil
+}
+
+// int64Check validates that within a single bundle,
+// we received the expected int64 values.
+// Returns ints, but they are unused, because we haven't
```
Comment (jrmccluskey) [inline sdks/go/pkg/beam/runners/direct/direct_test.go:142]:
```suggestion
// Returns strings, but they are unused, because we haven't
```

**Review by jrmccluskey — APPROVED** [review]
Looks like a good start to the test suite. LGTM mod typos!

**Comment by lostluck** [comment]
I'll definitely need to back update some of the other comments in the original, but I've removed the unrelated extras.

**Pull request merged**